run.py

```diff
@@ -0,0 +1,24 @@
+import yaml
+import random
+import torch
+
+def parse_yaml(f_path: str = 'config.yaml'):
+    with open(f_path, 'r') as f:
+        try:
+            args = yaml.safe_load(f)
+            return args
+        except yaml.YAMLError as exc:
+            print(exc)
```

Comment:
Can you add google style doc strings for every function you write. And also make sure you add type annotations for the return value

For now, just put this into chatgpt if you need guidance, but it should look like this 

def parse_yaml(f_path: str = 'config.yaml') -> dict[str, str]:
     """Parses yaml file containg training configurations
          
        Args:
            f_path: Path to config file

        Returns:
            A dictionary containing the training frameworks configs
     """

Something like that. There's more here because you have a try exception block (ask chatgpt)